src/thunderbots/software/ai/ai.cpp

```diff
@@ -1,6 +1,10 @@
 #include "ai.h"
 
-AI::AI() : navigator(std::make_unique<RRTNav>()), high_level(std::make_unique<STP_HL>())
+AI::AI()
+    : world(Field(), Ball(), Team(std::chrono::milliseconds(1000)),
+            Team(std::chrono::milliseconds(1000))),
```

Comment:
1 second seems a little arbitrary? Why not 0?

Reply:
Yes 1 second is a bit arbitrary, but this value also should (almost) never be 0, since that would mean if a robot was not detected for a single camera frame, it would be removed from the `Team` and re-added next frame, losing hysteresis and causing flickering or messed up assignments.

Regardless, I agree this could be handled better. I'm going to replace this `AI` constructor with one that takes parameters for the Ball, Teams etc. and also add a function to modify the `robot_expiry_buffer` for teams.

Reply:
> messed up assignments
The other side effects are fine and somewhat expected, but this seems scary. What do you mean?

Reply:
Currently no tracking for the removal of default constructors. The reason I didn't do that completely here is because if the default constructor is removed, 1 of 2 things must happen:
1. The declaration of the AI in `ai/main.cpp` must be marked as `extern` so that the object is not constructed immediately (instead constructed at definition/instantiation)
or
2. The AI must be constructed at the declaration (with all it's parameters)

I wanted to avoid using `extern` more as it may be confusing for people, and it also caused issues with linking, with the usages of the `ai` object in the callback functions since there was no definition of the `ai` variable in the "global" scope of the file. I also wanted to keep the initialization/construction of the `ai` object with the other variables in the beginning of the `main` function.

So if we want to remove default constructors (which I am totally ok with), we will need to solve this problem, or at least work around it in a more clean way. We might need to revise how any variables that need to maintain state in the main loop of the `main.cpp` file are stored. For example, creating a header file and keeping them there.

As for the "messed up assignments" side effect, that was poor wording on my part (although partially true). If we are defending the enemy team, and the number of robots on the field keeps changing rapidly and individual robots disappear for very short periods of time, our defenders may be rapidly assigned to different enemy robots because the threat keeps changing because of the flickering robots. This would result in us not "committing" to a decision (or likely making the correct one) and as a result not defending effectively. This is why the buffer value should be positive, so we can effectively "debounce" the detentions of robots and avoid this problem.

Reply:
Created #70 to track removal of default constructors

Reply:
All sounds good! 
- TIL what `extern` is for :laughing:. Totally fair reasoning.
- Ah, ok. Undesired behavior of the program is fine with me in this case, was more concerned about literally breaking the programatic constructs we're attempting to create (sorry, kinda tricky to phrase what I mean, but either way, all good).